python interface does not compile

This commit:

https://github.com/jfeist/scuff-em/tree/7ee9d73f988fedf8a1bb595dce8aa9a3a6db2635

seems to have broken the python build on several systems I am using, with the following error messages:

python.cpp:3572:34: error: 'NPY_ARRAY_C_CONTIGUOUS' was not declared in this scope
scuff-python.cpp:3572:59: error: 'NPY_ARRAY_ALIGNED' was not declared in this scope

This seems to be related to incompatibilities in the version of numpy.

--- a/src/libs/python/scuff-python.i
+++ b/src/libs/python/scuff-python.i
@@ -9,6 +9,15 @@
 %apply std::complex<double> { cdouble };
 
 %include "numpy.i" // numpy array conversions
+
+// some additional backward compatibility declarations for supporting numpy < 1.7.0
+%{
+#if NPY_API_VERSION < 0x00000007
+#define NPY_ARRAY_C_CONTIGUOUS NPY_C_CONTIGUOUS
+#define NPY_ARRAY_ALIGNED  NPY_ALIGNED
+#endif
+%}
+
 %init %{
   import_array();
 %}

